src/envoy/http/error_translator/filter.cc

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/envoy/http/error_translator/filter.h"
+
+#include <string>
+
+#include "absl/types/optional.h"
+#include "common/buffer/buffer_impl.h"
+#include "common/grpc/status.h"
+#include "common/http/headers.h"
+#include "src/envoy/utils/filter_state_utils.h"
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace error_translator {
+
+using ::Envoy::Http::FilterDataStatus;
+using ::Envoy::Http::FilterHeadersStatus;
+
+Filter::Filter(FilterConfigSharedPtr config) : config_(config) {}
+
+FilterHeadersStatus Filter::encodeHeaders(
+    Envoy::Http::ResponseHeaderMap& headers, bool) {
+  if (isUpstreamResponse()) {
+    // Do not translate any response sent from the upstream.
+    return FilterHeadersStatus::Continue;
+  }
+
+  if (headers.ContentType() != nullptr &&
+      headers.ContentType()->value().getStringView() ==
+          Envoy::Http::Headers::get().ContentTypeValues.Grpc) {
```

Comment:
we can call this function

https://cs.corp.google.com/piper///depot/google3/third_party/envoy/src/source/common/grpc/common.cc?type=cs&rcl=307518108&g=0&l=28

Reply:
Done